lib/shared/bucketing-assembly-script/assembly/types/dvcUser.ts

```diff
@@ -17,6 +26,29 @@ interface DVCUserInterface {
     privateCustomData: JSON.Obj | null
 }
 
+function getJSONObjFromPBCustomData(nullableCustomData: NullableCustomData | null): JSON.Obj | null  {
+    if (!nullableCustomData || nullableCustomData.isNull) return null
+
+    const customDataObj = new JSON.Obj()
+    const keys = nullableCustomData.value.keys()
+
+    for (let i = 0; i < keys.length; i++) {
+        const key = keys[i]
+        const value: CustomDataValue = nullableCustomData.value.get(key)
+        if (value && value.type === VariableType_PB.Boolean) {
+            customDataObj.set(key, value.boolValue)
+        } else if (value && value.type === VariableType_PB.Number) {
+            customDataObj.set(key, value.doubleValue)
+        } else if (value && value.type === VariableType_PB.String) {
+            customDataObj.set(key, value.stringValue)
+        } else {
+            throw new Error('DVCUser customData can\'t contain nested objects or arrays')
```

Comment:
what about null values?

Reply:
Nulls should be allowed - see current client-sdk not accepting nulls.

Reply:
correct. we want to allow explicitly setting a key's value to null

Reply:
that might be a little tricky to add, will take a look

Reply:
done.